dftimewolf/lib/processors/turbinia_gcp_tam.py

```diff
@@ -0,0 +1,361 @@
+          local paths, the second element contains the remote (GS) paths.
+    """
+    local_paths = []
+    gs_paths = []
+    for task in task_data:
+      # saved_paths may be set to None
+      saved_paths = task.get('saved_paths') or []
+      for path in saved_paths:
+
+        if path.endswith('.plaso') or \
+            path.endswith('BinaryExtractorTask.tar.gz') or \
+            path.endswith('hashes.json'):
+
+          if path.startswith('gs://'):
+            gs_paths.append(path)
+          else:
+            local_paths.append(path)
+
+    return local_paths, gs_paths
+
+  def _DownloadFilesFromGCS(self,
+                            timeline_label: str,
+                            gs_paths: List[str]) -> List[Tuple[str, str]]:
+    """Downloads files stored in Google Cloud Storage to the local filesystem.
+
+    Args:
+      timeline_label (str): Label to use to construct the path list.
+      gs_paths (List[str]):  gs:// URI to files that need to be downloaded
+          from GS.
+
+    Returns:
+      list:
+        tuple: containing:
+          str: The timeline label for this path.
+          str: A local path where GS files have been copied to.
+    """
+    # TODO: Externalize fetching files from GCS buckets to a different module.
+
+    local_paths = []
+    for path in gs_paths:
+      local_path = None
+      try:
+        output_writer = output_manager.GCSOutputWriter(
+            path, local_output_dir=self._output_path)
+        local_path = output_writer.copy_from(path)
+      except TurbiniaException as exception:
+        # Don't add a critical error for now, until we start raising errors
+        # instead of returning manually each
+        self.ModuleError(str(exception), critical=False)
+      self.logger.success('Downloaded {0:s} to {1:s}'.format(path, local_path))
+
+      if local_path:
+        local_paths.append((timeline_label, local_path))
+
+    return local_paths
+
+  def TurbiniaSetUp(self,
+                    project: str,
+                    turbinia_zone: str,
+                    sketch_id: int,
+                    run_all_jobs: bool) -> None:
+    """Sets up the object attributes.
+
+    Args:
+      project (str): name of the GCP project containing the disk to process.
+      turbinia_zone (str): GCP zone in which the Turbinia server is running.
+      sketch_id (int): The Timesketch sketch ID.
+      run_all_jobs (bool): Whether to run all jobs instead of a faster subset.
+    """
+    self.project = project
+    self.turbinia_zone = turbinia_zone
+    self.sketch_id = sketch_id
+    self.run_all_jobs = run_all_jobs
+
+    turbinia_config.LoadConfig(config_file=self.turbinia_config_file)
+    if not self.project:
+      self.project = turbinia_config.TURBINIA_PROJECT
+    if not self.turbinia_zone:
+      self.turbinia_zone = turbinia_config.TURBINIA_ZONE
+
+    if self.project is None or self.turbinia_zone is None:
```

Comment:
If I'm reading this correctly, the empty string will skip that safeguard. However seeing the error that's thrown below it sounds like an empty string would also break things. Is that correct? If so, consider using `if not self.project or not self.turbinia_zone` instead.